java/debugger/debugger-agent/src/com/intellij/rt/debugger/agent/CaptureAgent.java

```diff
@@ -617,7 +617,9 @@ private static KeyProvider param(int idx) {
     addCapture("akka/actor/ScalaActorRef", "$bang", FIRST_PARAM);
     addCapture("akka/actor/RepointableActorRef", "$bang", FIRST_PARAM);
     addCapture("akka/actor/LocalActorRef", "$bang", FIRST_PARAM);
+    addCapture("akka/actor/ActorRef", "$bang", FIRST_PARAM);
     addInsert("akka/actor/Actor$class", "aroundReceive", param(2));
+    addInsert("akka/actor/Actor", "aroundReceive", param(2));
```

Comment:
How are static / non-static methods handled? Is `param` zero-based?

In the trait implementation class `Actor$class`, the method `aroundReceive` is static. As you say, it has a `$this` parameter, the actor parameter is the second.

When compiled with 2.12+, the `aroundReceive` method is now a default method in `Actor`, it's no longer static. My assumption was that `param(1)` would be the recevier `this`, that's why I chose `param(2)`. So that is not correct?

Let me add that I didn't test this PR in any way!

Reply:
`param` is zero-based, and there is a `THIS_KEY_PROVIDER` to access the receiver. You may see that `ParamKeyProvider.loadKey` skips the first local variable for non-static methods.

Reply:
OK, thank you!